### PR TITLE
Many changes

### PR DESCRIPTION
**Pull request by wherkamp**
This is just a redo of the mod loading process.

### CODE REVIEW

File: src/main/java/org/kakara/core/game/GameEventManager.java [hunk 1]
```diff
@@ -25,7 +26,7 @@ public void registerEventHandler(Object handler, Mod mod) {
 
             Class<?> param = method.getParameters()[0].getType();
             if (!param.isAssignableFrom(Event.class) || method.getParameters().length != 1) {
-                KakaraCore.LOGGER.warn(String.format("Unable to register %s dulse to not taking an event.", method.getName()));
+                Kakara.LOGGER.warn(String.format("Unable to register %s dulse to not taking an event.", method.getName()));
```
Comment (Monology) [inline src/main/java/org/kakara/core/game/GameEventManager.java]: What is a dulse?

Comment (wherkamp) [reply src/main/java/org/kakara/core/game/GameEventManager.java]: I have no freaking clue.

File: src/main/java/org/kakara/core/mod/Mod.java [hunk 2]
```diff
@@ -20,5 +21,8 @@
 
     Logger getLogger();
 
-    KakaraCore getKakaraCore();
+    GameInstance getGameInstance();
+
+    void preEnable();
+    void postEnable();
```
Comment (Monology) [inline src/main/java/org/kakara/core/mod/Mod.java]: just make it say beforeEnable and afterEnable, this is english

**Comment by wherkamp** [comment]
This solved #8

**Pull request merged**